sqlmesh/core/engine_adapter/databricks.py

```diff
@@ -1,13 +1,112 @@
+from sqlmesh.utils import classproperty
+
+if t.TYPE_CHECKING:
+    from sqlmesh.core.engine_adapter._typing import DF, PySparkSession
 
+logger = logging.getLogger(__name__)
 
-class DatabricksSparkSessionEngineAdapter(SparkEngineAdapter):
+
+class DatabricksEngineAdapter(SparkEngineAdapter):
     DIALECT = "databricks"
     SCHEMA_DIFFER = SchemaDiffer(
         support_positional_add=True,
         support_nested_operations=True,
         array_element_selector="element",
     )
+
+    def __init__(
+        self,
+        connection_factory: t.Callable[[], t.Any],
+        dialect: str = "",
+        sql_gen_kwargs: t.Optional[t.Dict[str, Dialect | bool | str]] = None,
+        multithreaded: bool = False,
+        **kwargs: t.Any,
+    ):
+        super().__init__(
+            connection_factory,
+            dialect,
+            sql_gen_kwargs,
+            multithreaded,
+            **kwargs,
+        )
+        self._spark: t.Optional[PySparkSession] = None
+
+    @classproperty
+    def can_access_spark_session(cls) -> bool:
+        from sqlmesh import runtime_env
+
+        if runtime_env.is_databricks:
+            return True
+        try:
+            from databricks.connect import DatabricksSession  # type: ignore
+
+            return True
+        except ImportError:
+            return False
+
+    @property
+    def _use_spark_session(self) -> bool:
+        from sqlmesh import runtime_env
+
+        if runtime_env.is_databricks:
+            return True
+        return self.can_access_spark_session and {
+            "databricks_connect_server_hostname",
+            "databricks_connect_access_token",
+            "databricks_connect_cluster_id",
+        }.issubset(self._extra_config)
+
+    @property
+    def is_spark_session_cursor(self) -> bool:
+        from sqlmesh.engines.spark.db_api.spark_session import SparkSessionCursor
+
+        return isinstance(self.cursor, SparkSessionCursor)
+
+    @property
+    def spark(self) -> PySparkSession:
+        if self.is_spark_session_cursor:
+            return self._connection_pool.get().spark
+
+        from databricks.connect import DatabricksSession
+
+        if self._spark is None:
+            self._spark = DatabricksSession.builder.remote(
+                host=self._extra_config["databricks_connect_server_hostname"],
+                token=self._extra_config["databricks_connect_access_token"],
+                cluster_id=self._extra_config["databricks_connect_cluster_id"],
+            ).getOrCreate()
+            catalog = self._extra_config.get("catalog")
+            if catalog:
+                self._spark.sql(f"USE CATALOG {catalog}")
+        return self._spark
+
+    def _fetch_native_df(self, query: t.Union[exp.Expression, str]) -> DF:
+        """Fetches a DataFrame that can be either Pandas or PySpark from the cursor"""
+        if self.is_spark_session_cursor:
+            return super()._fetch_native_df(query)
+        if self.can_access_spark_session:
```

Comment:
Why do we sometimes use this and sometimes we use `_use_spark_session ` instead? Having both seem to be redundant. For example in which case do I have access to the spark session but don't want to use it?

Reply:
So the scenario is that someone has Databricks Connect installed but doesn't want to use it. So they want everything routed through the SQL connector which means converting pandas to SQL. 

At one point I had `disable_databricks_connect` as a property but removed it since I was curious if this was really needed. I see though that without this then this property doesn't make sense. So I think I'm going to add it back in.